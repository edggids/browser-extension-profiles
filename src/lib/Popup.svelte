<script lang="ts">
    export let isOpen = false;
    let className = '';

    export { className as class};
</script>

<div role="dialog" class={`${className} ${isOpen ? '': 'hidden'} p-2 bg-ground rounded shadow z-10`}>
    <slot></slot>
</div>

