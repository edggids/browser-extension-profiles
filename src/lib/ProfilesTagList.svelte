<script lang="ts">
    import Tag from "./Tag.svelte";
    import {createEventDispatcher} from "svelte";

    export let profiles = [];
    export let currentProfileId;

    const dispatch = createEventDispatcher();
    const onClick = (id) => dispatch("selected", { id });
</script>

{#each profiles as [id, profile]}
    <Tag active={id === currentProfileId } on:click={e => onClick(id)} class="mr-4 mb-4" color={profile.color.default}>{profile.name}</Tag>
{/each}
