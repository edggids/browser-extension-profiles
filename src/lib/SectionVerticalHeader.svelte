<script lang="ts">
let className = "";
export {className as class};

export let title = "";
export let compact = false;

const paddings = compact ? "pt-4 pb-5" : "pt-6 pb-2.75";
</script>

<section class={`relative ${className}`}>
    <span data-testid="title" class="vertical-header text-xs uppercase transition-color text-current">{title}</span>

    <div data-testid="contentWrapper" class={`container transition-bg bg-current-light ${paddings}`}>
        <slot></slot>
    </div>
</section>
