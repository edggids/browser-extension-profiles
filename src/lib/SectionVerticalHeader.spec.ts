import {cleanup, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach} from 'vitest';
import SectionVerticalHeader from "./SectionVerticalHeader.svelte";
import Popup from "./Popup.svelte";
import ProfilesTagList from "./ProfilesTagList.svelte";

describe('SectionVerticalHeader component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container } = render(SectionVerticalHeader, { props: { title: 'Test Title' } });

    expect(container).toMatchSnapshot();
  });

  it('renders with provided title', () => {
    const { container, getByTestId } = render(SectionVerticalHeader, { props: { title: 'Test Title' } });
    const header = getByTestId('title');
    expect(header.textContent).toBe('Test Title');
  });

  it('applies custom class when provided', () => {
    const { container } = render(SectionVerticalHeader, { props: { compact: false, class: 'custom-class' } });
    const section = container.querySelector('section');

    expect(section.classList.contains('custom-class')).toBe(true);
  });

  it('renders with default padding when compact is false', () => {
    const {  getByTestId, container } = render(SectionVerticalHeader, { props: { compact: false } });
    const target = getByTestId('contentWrapper');

    expect(target.classList.contains('pt-6')).toBe(true);
    expect(target.classList.contains('pb-2.75')).toBe(true);
  });

  it('renders with compact padding when compact is true', () => {
    const {  getByTestId, container } = render(SectionVerticalHeader, { props: { compact: true } });
    const target = getByTestId('contentWrapper');

    expect(target.classList.contains('pt-4')).toBe(true);
    expect(target.classList.contains('pb-5')).toBe(true);
  });
});
