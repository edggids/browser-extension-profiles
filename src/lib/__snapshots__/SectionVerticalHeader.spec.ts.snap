// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionVerticalHeader component > mounts 1`] = `
<body>
  <div>
    <section
      class="relative "
    >
      <span
        class="vertical-header text-xs uppercase transition-color text-current"
        data-testid="title"
      >
        Test Title
      </span>
       
      <div
        class="container transition-bg bg-current-light pt-6 pb-2.75"
        data-testid="contentWrapper"
      />
    </section>
  </div>
</body>
`;
